lib/Onyx.js

```diff
@@ -874,8 +891,19 @@ function set(key, value) {
         return remove(key);
     }
 
+    if (hasPendingMergeForKey(key)) {
+        Logger.logAlert(`Onyx.set() called after Onyx.merge() for key: ${key}. It is recommended to use set() or merge() not both.`);
+    }
+
+    const hasChanged = cache.hasValueChanged(key, value);
+
     // This approach prioritizes fast UI changes without waiting for data to be stored in device storage.
-    broadcastUpdate(key, value, 'set');
+    broadcastUpdate(key, value, hasChanged, 'set');
+
+    // If the value has not changed then we don't need to write it to storage
```

Comment:
This comment should explain _why_ and it currently only explains _what_. Why doesn't it need to be written to storage?

> If the value has not changed, calling Storage.setItem() would be redundant and a waste of performance, so return early instead.